pkg/testdatagen/scenario/devseed.go

```
@@ -408,6 +410,103 @@ func createMoveWithPPMAndHHG(db *pop.Connection, userUploader *uploader.UserUplo
+		},
+	})
+
+	estimatedHHGWeight := unit.Pound(1400)
+	actualHHGWeight := unit.Pound(2000)
+	testdatagen.MakeMTOShipment(db, testdatagen.Assertions{
+		MTOShipment: models.MTOShipment{
+			ID:                   uuid.FromStringOrNil("8689afc7-84d6-4c60-a739-8cf96ede2601"),
+			PrimeEstimatedWeight: &estimatedHHGWeight,
+			PrimeActualWeight:    &actualHHGWeight,
+			ShipmentType:         models.MTOShipmentTypeHHG,
+			ApprovedDate:         swag.Time(time.Now()),
+			Status:               models.MTOShipmentStatusSubmitted,
+			MoveTaskOrder:        move,
+			MoveTaskOrderID:      move.ID,
+		},
+	})
+
+	testdatagen.MakeMTOShipment(db, testdatagen.Assertions{
+		MTOShipment: models.MTOShipment{
+			ID:                   uuid.FromStringOrNil("8689afc7-84d6-4c60-a739-333333333331"),
+			PrimeEstimatedWeight: &estimatedHHGWeight,
+			PrimeActualWeight:    &actualHHGWeight,
+			ShipmentType:         models.MTOShipmentTypeHHG,
+			ApprovedDate:         swag.Time(time.Now()),
+			Status:               models.MTOShipmentStatusSubmitted,
+			MoveTaskOrder:        move,
+			MoveTaskOrderID:      move.ID,
+		},
+	})
+
+	testdatagen.MakeMTOShipment(db, testdatagen.Assertions{
+		MTOShipment: models.MTOShipment{
+			PrimeEstimatedWeight: &estimatedHHGWeight,
+			PrimeActualWeight:    &actualHHGWeight,
+			ShipmentType:         models.MTOShipmentTypeHHG,
+			ApprovedDate:         swag.Time(time.Now()),
+			Status:               models.MTOShipmentStatusRejected,
+			MoveTaskOrder:        move,
+			MoveTaskOrderID:      move.ID,
+		},
+	})
+
+	ppm := testdatagen.MakePPM(db, testdatagen.Assertions{
+		ServiceMember: move.Orders.ServiceMember,
+		PersonallyProcuredMove: models.PersonallyProcuredMove{
+			OriginalMoveDate: &nextValidMoveDate,
+			Move:             move,
+			MoveID:           move.ID,
+		},
+		UserUploader: userUploader,
+	})
+
+	move.PersonallyProcuredMoves = models.PersonallyProcuredMoves{ppm}
+	move.Submit(time.Now())
+	models.SaveMoveDependencies(db, &move)
```

These will get flagged in static analysis if we don't check the return value for errors I think although I see it's not checked elsewhere in the file.  I've also seen this fail silently and then the test data is not what we think it is.

---

Did you want multiple shipments, as well as a PPM move on purpose? If not, if you just want a move whose Order doesn't have a TAC, I think this can be simplified to this:
```golang
move := testdatagen.MakeHHGMoveWithShipment(db, testdatagen.Assertions{
    Move: models.Move{
        Locator: "NILTAC",
   },
})
order := move.Orders
order.TAC = nil
verrs, err := models.SaveMoveDependencies(db, &move)
if verrs.HasAny() || err != nil {
    logger.Error("errors encountered saving move and its dependencies", zap.Error(err))
}
```

---

This move is not appearing in the TOO queue. When I look at it in TablePlus, I can see that it still has a status of DRAFT. How did you test this locally? I don't see any other orders with missing information, so I'm confused as to how y'all tested this PR.

---

I fixed it in my PR, and I also made sure to remove all the other required fields to make it easier to test that you can't approve the Move with one missing field at a time.

---

My bad. The confusion was because I was testing this on my branch, where the Orders validation caused the `SaveMoveDependencies` to silently fail and didn't create this move. I apologize if my tone sounded accusatory.